### PR TITLE
feat: Add option to allow unused types

### LINKED ISSUE #420: Unused type definitions are not visible in introspection queries
If we define a type that is unused in the schema file e.g

```graphql
type User {
    ...
}

type Context {
    ...
}

type Query {
    user(id: ID!): User
}
```

- The `Query` and `User` types are returned from introspection queries but not `Context`. Is this by design?
- If it is by design, is there a way for the type to appear in introspection queries? I can see they appear in `SchemaParser.getUnusedDefinitions()` but is there a way to wire that in so that they are visible in introspection queries?

### LINKED ISSUE #442: Federated Entity type removed from executable schema
## Background & Description

First of all, thank you all for the great lib! It is great to see such a mature Schema First library for GraphQL in Java, I'm really enjoying working with it.

I'm trying to use `graphql-java-tools` with [federation-jvm](https://github.com/apollographql/federation-jvm) and overall it works great.

However, I'm facing one issue where we have a service which defines its own query, but also extends one entity to add fields to it. Like the schema [here](https://github.com/danilo-barboza/apollo-federation-java-tools/blob/master/src/main/resources/schema.graphqls).

The root of the issue is that `graphql-java-tools` ignores the `User` type because it is "orphan" as it is not referenced by any other type in the schema. However, this is a completely valid scenario, and would be great if `graphql-java-tools` could support it.

## How to reproduce

I've created a simple project reproducing the issue: https://github.com/danilo-barboza/apollo-federation-java-tools/

Just clone it and run

```
./gradlew test
```

## Possible solutions

I've also hacked hard and created a draft Pull Request to show a possible way to fix it: #441. The main idea is to make sure we add types that have the `@extends` annotation.

### PR DESCRIPTION
**Pull request by danilo-barboza**
Resolves #442
Resolves #420 

## Checklist
- [x] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [x] New or modified functionality is covered by tests

## Description
Adding a new `includeUnusedTypes` option to include add unused types to the final executable schema. The caveat is that the unused types need to be included in the schema parser dictionary. I think it is an ok tradeoff as this feature has specific use cases as described in #420 and #442.

Added a couple of test cases to it as well as serve as both test and example of how to use it.

### CODE REVIEW

**Comment by danilo-barboza** [comment]
Hi @vojtapol.

Can you take a look at this approach? Is it valid? What do you think?

Looking forward to getting your feedback.

**Comment by barnardb** [comment]
This looks like a straightforward way to expose the schema as written and unblock the ability to extend a type in the federation use case. Looking forward to seeing this get a maintainer review. :)

**Review by vojtapol — APPROVED** [review]
Thank you so much for the contribution. It has no impact on existing users whatsoever which is great.

**Comment by danilo-barboza** [comment]
Thank you @vojtapol for merging the PR!

What is the release process? When should we expect a new version with the changes to be released?

**Comment by vilkazz** [comment]
+1 on the release date. Would love to have access to this feature

**Pull request merged**
